py-polars/tests/unit/io/test_delta.py

```diff
@@ -179,3 +179,175 @@ def test_write_delta(df: pl.DataFrame, tmp_path: Path) -> None:
     assert df_supported.columns == pl_df_partitioned.columns
 
     df_supported.write_delta(partitioned_tbl_uri, mode="overwrite")
+
+
+@pytest.mark.write_disk()
+def test_write_delta_w_compatible_schema(tmp_path: Path) -> None:
+    import datetime
+
+    from deltalake import DeltaTable
+
+    from polars import Datetime, Field, List, Struct, UInt32, Utf8
+
+    df = pl.DataFrame(
+        [
+            pl.Series("string", ["test"], dtype=pl.Utf8),
```

Comment:
I think these can all be separate test cases instead of one big dataframe. That helps determine which data type fails the test. You can use `@pytest.mark.parametrize` for this.

Reply:
I've parametrized the tests now, each column is a new test case.